src/main/java/org/checkerframework/checker/noliteral/NoLiteralVisitor.java

```diff
@@ -0,0 +1,230 @@
+   *
+   * <p>Stolen shamelessly from https://github.com/awslabs/data-classification-checker
+   *
+   * @return a singleton set containing the @MaybeConstant annotation
+   */
+  @Override
+  protected Set<? extends AnnotationMirror> getThrowUpperBoundAnnotations() {
+    return Collections.singleton(atypeFactory.getCanonicalTopAnnotation());
+  }
+
+  /**
+   * Searches through a type for user-written (i.e. non-default) annotations. Returns true if any
+   * are found. Stolen shamelessly from https://github.com/awslabs/data-classification-checker
+   *
+   * @param atm the type to search
+   * @return true if the type or any of its components has a non-default annotation
+   */
+  private boolean hasNonDefault(final AnnotatedTypeMirror atm) {
+    boolean result = false;
+    switch (atm.getKind()) {
+      case DECLARED:
+        AnnotatedTypeMirror.AnnotatedDeclaredType atmD =
+            (AnnotatedTypeMirror.AnnotatedDeclaredType) atm;
+        for (AnnotatedTypeMirror component : atmD.getTypeArguments()) {
+          if (hasNonDefault(component)) {
+            result = true;
+          }
+        }
+        break;
+      case ARRAY:
+        AnnotatedTypeMirror.AnnotatedArrayType atmA = (AnnotatedTypeMirror.AnnotatedArrayType) atm;
+        result = hasNonDefault(atmA.getComponentType());
+        break;
+      case WILDCARD:
+        return false;
+      case TYPEVAR:
+        return false;
+      default:
+        break;
+    }
+    return result || !atm.hasAnnotation(NonConstant.class);
+  }
+
+  /**
+   * This checker defaults implicit upper bounds of type variables to @NonConstant. This is the
+   * correct thing to do in unannotated code, but causes false positive errors whenever a user
+   * writes a type annotation on the upper bound of a type variable (for instance, by declaring a
+   * list of constant strings). Since this is common, this code disables that check for user-written
+   * annotations. Stolen shamelessly from https://github.com/awslabs/data-classification-checker
+   */
+  @Override
+  protected void checkTypeArguments(
+      final Tree toptree,
+      final List<? extends AnnotatedTypeParameterBounds> paramBounds,
+      final List<? extends AnnotatedTypeMirror> typeargs,
+      final List<? extends Tree> typeargTrees) {
+    List<AnnotatedTypeParameterBounds> newParamBounds = new ArrayList<>();
+    List<AnnotatedTypeMirror> newTypeArgs = new ArrayList<>();
+    for (int i = 0; i < typeargs.size(); i++) {
+      AnnotatedTypeMirror atm = typeargs.get(i);
+      if (!hasNonDefault(atm)) {
+        newParamBounds.add(paramBounds.get(i));
+        newTypeArgs.add(atm);
+      }
+    }
+    if (!newTypeArgs.isEmpty()) {
+      super.checkTypeArguments(toptree, newParamBounds, newTypeArgs, typeargTrees);
+    }
+  }
+
+  /**
+   * Skip the standard subtyping check on method calls always. Constants can't have methods called
+   * on them, so we don't care.
```

Comment:
Some constants, such as strings, can have methods called on them.  I would expect String methods to be polymorphic in their arguments.  I don't see any test cases for string method calls.

I was surprised not to find a polymorphic annotation in this type system, since it is useful for most type systems.